xds/src/test/java/io/grpc/xds/XdsLoadBalancerTest.java

```diff
@@ -91,54 +96,52 @@ public LoadBalancer newLoadBalancer(Helper helper) {
     }
   };
 
-
-
   @Before
   public void setUp() {
     MockitoAnnotations.initMocks(this);
-    LoadBalancerRegistry.getDefaultRegistry().register(lbProvider1);
-    LoadBalancerRegistry.getDefaultRegistry().register(lbProvider2);
-    lb = new XdsLoadBalancer(helper);
-  }
-
-  @After
-  public void tearDown() {
-    LoadBalancerRegistry.getDefaultRegistry().deregister(lbProvider1);
-    LoadBalancerRegistry.getDefaultRegistry().deregister(lbProvider2);
+    lbRegistry.register(lbProvider1);
+    lbRegistry.register(lbProvider2);
+    lb = new XdsLoadBalancer(helper, lbRegistry);
+    doReturn(new SynchronizationContext(Thread.currentThread().getUncaughtExceptionHandler()))
```

Comment:
You probably want to create the exception handler like you did in FallbackManagerTest.